### ISSUE TITLE
Generating Mlflow graphs with all runs across checkpoints stitched together.

### ISSUE THREAD

**Issue by artificiallama** [ecmwf/anemoi-core#1321]
I ran two training experiments.

run1 goes from epoch 1 to 5.
run2 which uses the last checkpoint from run1, goes from epoch 5 to epoch 12.

I can see the graphs corresponding to each of the two runs such as training loss. Is there any way in Mlflow to display such graphs
with the two runs appended together ? In this case, I would like to see the graph of training loss with xaxis going from epoch number 1 to 12 ?

I am using,

```
mlflow==2.22.5
mlflow-skinny==2.22.5
```
I had checked out anemoi-core on 22 july 2026.

```
anemoi-datasets==0.5.40
-e git+https://github.com/ecmwf/anemoi-core.git@6278cf06c47f8d226ecad857bbc3185b0486724d#egg=anemoi_graphs&subdirectory=graphs
-e git+https://github.com/ecmwf/anemoi-core.git@6278cf06c47f8d226ecad857bbc3185b0486724d#egg=anemoi_models&subdirectory=models
-e git+https://github.com/ecmwf/anemoi-core.git@6278cf06c47f8d226ecad857bbc3185b0486724d#egg=anemoi_training&subdirectory=training
anemoi-transform==0.4.3
anemoi-utils==0.5.9
```

Thanks.

**Comment by mchantry** [comment]
@artificiallama thanks for the issue, could you share the commands you use to restart please, as this can affect how the mlflow data is aggregated.

**Comment by artificiallama** [comment]
In the yaml file under `training/` I have set,

```
run_id: 335201fbc9d148449063955a140099e4
fork_run_id: null
transfer_learning: False # activate to perform transfer learning
load_weights_only: False # only load model weights, do not restore optimiser states etc.
update_ds_stats_on_ckpt_load:
  states: False # rebuild state processors from current dataset when loading a checkpoint
  tendencies: True # rebuild tendency processors from current dataset when loading a checkpoint
```

I understand that setting `run_id` to SHA of the checkpoint and `fork_run_id` to null tells anemoi that I intend to continue the same training run restarting from the checkpoint. It seems to me that Mlflow is picking this up correctly since  the Mlflow page shows,


<img width="800" height="60" alt="Image" src="https://github.com/user-attachments/assets/029d4bd3-2792-4f93-86c4-68c67bc96c20" />


As you can see in the above image, the restarted run appears "under" the run that produced the checkpoint. I have set the `run_name` in
the `mlflow.yaml` same for both the runs. Thanks.

**Comment by gabrieloks** [comment]
Hello!

On the mlflow page, if you want to plot training info like loss curves etc for a given training, or to compare different trainings you need to click on the experiments:

<img width="378" height="83" alt="Image" src="https://github.com/user-attachments/assets/f1a7853b-5d8c-4534-a3e9-f7d6b760808b" />

And then you click on compare:

<img width="378" height="83" alt="Image" src="https://github.com/user-attachments/assets/638262ba-25a9-4a75-a218-2965dce33630" />

go down to metrics:

<img width="505" height="324" alt="Image" src="https://github.com/user-attachments/assets/1842a4d2-7099-42d0-b34d-21694a79667f" />

and then you click on train_loss_epoch for 

<img width="1219" height="525" alt="Image" src="https://github.com/user-attachments/assets/cdf78f5b-5f37-4712-b2a6-87892ab0493f" />

Hope this helps!

Gabriel

**Comment by artificiallama** [comment]
Thanks. This worked for me. To obtain appended curves for checkpointed runs on the left side panel choose "steps" radio button for X-axis.

<img width="1158" height="511" alt="Image" src="https://github.com/user-attachments/assets/982b88f3-a062-4664-ab5e-9d6e3fcc898b" />

I have a question though - as you can see there a gap between the two curves at the checkpoint. Is this normal ? I see this gap in the graph you have posted as well !  I guess this is because the loss is logged at **the end** of every epoch. The appending does not extrapolate between the last checkpointed epoch and the first epoch loss value of the new (second) training run ?
Thanks.
